auditbeat/module/audit/file/eventreader_fsnotify.go

```diff
@@ -0,0 +1,119 @@
+// +build linux freebsd openbsd netbsd windows darwin
+
+package file
+
+import (
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/fsnotify/fsnotify"
+)
+
+// NewEventReader creates a new EventReader backed by fsnotify.
+func NewEventReader(c Config) (EventReader, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	return &reader{watcher, c, make(chan Event, 1), make(chan error, 1)}, nil
+}
+
+type reader struct {
+	watcher *fsnotify.Watcher
+	config  Config
+	outC    chan Event
+	errC    chan error
+}
+
+func (r *reader) Start(done <-chan struct{}) (<-chan Event, error) {
+	var paths []string
+	for _, filePaths := range r.config.Paths {
+		paths = append(paths, filePaths...)
+	}
+
+	for _, p := range paths {
+		if err := r.watcher.Add(p); err != nil {
+			if err == syscall.EMFILE {
+				logp.Warn("%v Failed to watch %v: %v (check the max number of "+
+					"open files allowed with 'ulimit -a')", logPrefix, p, err)
+			} else {
+				logp.Warn("%v Failed to watch %v: %v", logPrefix, p, err)
+			}
+		}
+	}
+
+	go func() {
+		defer close(r.outC)
+		defer r.watcher.Close()
+
+		for {
+			select {
+			case event := <-r.watcher.Events:
+				if event.Name == "" {
+					continue
+				}
+				r.outC <- convertToFileEvent(event, r.config.MaxFileSize)
+			case err := <-r.watcher.Errors:
+				r.errC <- err
+			}
+		}
+	}()
+
+	return r.outC, nil
+}
+
+func convertToFileEvent(e fsnotify.Event, maxFileSize int64) Event {
+	event := Event{
+		Timestamp: time.Now().UTC(),
```

Comment:
That [reminds me](https://github.com/elastic/beats/issues/4569), I think I will open a PR to automatically convert all time.Time objects to UTC during event normalization. Then we won't need a helper because it won't matter.